test/Sema/enum_raw_representable.swift

```diff
@@ -204,8 +204,7 @@ func sr8150_mutable(obj: SR8150Box) {
     sr8150_helper1(opt)
     // expected-error@-1 {{cannot convert value of type 'Bar?' to expected argument type 'Double'}} {{23-23=?.rawValue ?? <#default value#>}}
     sr8150_helper1(opt ?? Bar.a)
-    // expected-error@-1 {{no exact matches in call to global function 'sr8150_helper1'}}
-    // expected-note@-2  {{candidate expects value of type 'Bar' for parameter #0}} {{20-20=(}} {{32-32=).rawValue}}
+    // expected-error@-1 {{cannot convert value of type 'Bar' to expected argument type 'Double'}} {{20-20=(}} {{32-32=).rawValue}}
```

Comment:
Should this produce 2 error messages/a fix-it for each side of `??`? Also, how is it determined that the overload accepting `Double` is preferred because of the raw value?

Reply:
I think it's correct that it suggests `Double(opt ?? Bar.a).rawValue` now since that's the best possible fix. This change is due to "local ambiguity" fix where before `OverloadDiff` would make `diagnoseAmbiguity` confused since it would have more overloads in a diff than `aggregateFix` had fixes, now we completely ignore `sr8150_helper1(_: Int)` since it's a generic failed conversion fix.

Reply:
Oohhh I didn't realize that the fix-it was wrapping the result of `??` in `.rawValue`. That makes a lot more sense. Agreed that this is the best possible fix, thanks for the explanation!